src/components/OneToDo/OneToDo.jsx

```diff
@@ -2,30 +2,77 @@ import { Checkbox, IconButton, TextField, Typography } from "@mui/material";
 import DeleteIcon from "@mui/icons-material/Delete";
 import EditIcon from "@mui/icons-material/Edit";
 import SaveIcon from "@mui/icons-material/Save";
-import React, { useRef } from "react";
+import React, { useEffect, useRef } from "react";
 import { useState } from "react";
 import "./OneToDo.css";
+import { useToDoList, editToDo, deleteToDo } from "../../store/store";
+import { useForm } from "react-hook-form";
+import { useMutation } from "@tanstack/react-query";
+import { deleteToDoFromJSON, putToDoInJSON } from "../../api/todoAPIs";
+
+function OneToDo({ index }) {
+  const editMutation = useMutation({
+    mutationFn: (editedToDo) => {
+      return putToDoInJSON({ id: editedToDo.id, newToDo: editedToDo });
+    },
+    mutationKey: ["editToDo"],
+  });
+
+  const deleteMutation = useMutation({
+    mutationKey: ["deleteToDo"],
+    mutationFn: (id) => deleteToDoFromJSON(id),
+  });
 
-function OneToDo({ index, toDoList, editToDo, deleteToDo }) {
   const [editingToDo, setEditingToDo] = useState(false);
   const [toDoStatus, setToDoStatus] = useState(false);
 
   const editField = useRef();
 
+  const toDoList = useToDoList().toDoList;
+
+  useEffect(() => {
+    setToDoStatus(!toDoList[index].checked);
+  }, []);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const onSubmit = (data, e) => {
+    if (editingToDo === false) {
+      setEditingToDo(true);
+    } else {
+      setEditingToDo(false);
+      const editedToDo = {
+        text: data.ToDoField,
+        checked: toDoList[index].checked,
+        id: toDoList[index].id,
+      };
+      editToDo(editedToDo);
+      editMutation.mutate(editedToDo);
+    }
+  };
+
   const editToDoHandler = () => {
-    editToDo({
+    const editedToDo = {
       text: editField.current.value,
+      checked: toDoStatus,
```

Comment:
you don't need to send the status on each edit operation just send the text and id and inside 
your editTodo controller function use the spread operator to get the status value

Reply:
similary inside your putToDoInJSON() api function make put request to json server to only update the text of existing toto